odbc/src/api/free_stmt_tests.rs

```diff
@@ -0,0 +1,137 @@
+        let stmt = create_stmt_handle();
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLFreeStmt(stmt as *mut _, FreeStmtOption::SQL_RESET_PARAMS as i16)
+            )
+        }
+    }
+
+    #[test]
+    fn test_free_stmt_unbind() {
+        let env = &mut MongoHandle::Env(Env::with_state(EnvState::Allocated));
+        let conn =
+            &mut MongoHandle::Connection(Connection::with_state(env, ConnectionState::Allocated));
+        let stmt: *mut _ =
+            &mut MongoHandle::Statement(Statement::with_state(conn, StatementState::Allocated));
+
+        unsafe {
+            // Set the bound_cols to non-None initially.
+            let s = (*stmt).as_statement().unwrap();
+            *s.bound_cols.write().unwrap() = Some(map! {
+                1 => BoundColInfo {
+                    target_type: 1,
+                    target_buffer: null_mut(),
+                    buffer_length: 1,
+                    length_or_indicator: null_mut(),
+                }
+            });
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLFreeStmt(stmt as *mut _, FreeStmtOption::SQL_UNBIND as i16)
+            );
+
+            // Assert that the bound_cols are None after SQLFreeStmt is called.
+            assert_eq!(None, *s.bound_cols.read().unwrap())
+        }
+    }
+
+    #[test]
+    fn test_free_stmt_close() {
+        let env = &mut MongoHandle::Env(Env::with_state(EnvState::Allocated));
+        let conn =
+            &mut MongoHandle::Connection(Connection::with_state(env, ConnectionState::Allocated));
+        let stmt: *mut _ =
+            &mut MongoHandle::Statement(Statement::with_state(conn, StatementState::Allocated));
+
+        unsafe {
+            // Set the mongo_statement to have non-empty cursor initially.
+            // Here, we create a MockQuery with nonsense dummy data since the
+            // values themselves do not matter.
+            let mock_query = &mut MongoQuery::new(
+                vec![doc! {"x": "y"}, doc! {"x": "z"}],
+                vec![MongoColMetadata::new(
+                    "test_db",
+                    "dn".to_string(),
+                    "fn".to_string(),
+                    Schema::Atomic(Atomic::Scalar(BsonTypeName::String)),
+                    Nullability::SQL_NO_NULLS,
+                    TypeMode::Simple,
+                )],
+            );
+
+            // Must call next to set the `current` field.
+            let _ = mock_query.next(None);
+
+            // Set the mongo_statement
+            let s = (*stmt).as_statement().unwrap();
+            *s.mongo_statement.write().unwrap() = Some(Box::new(mock_query.clone()));
+
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLFreeStmt(stmt as *mut _, FreeStmtOption::SQL_CLOSE as i16)
+            );
+
+            // Assert that the mongo_statement closed the cursor (no resultset, no current).
```

Comment:
I didn't want to make the `MockQuery` fields public, so instead I am indirectly testing the "closed cursor" state here by first trying to `get_data` (which, if not for `SQLFreeStmt`, would be valid since we already called `next` earlier) and then trying to call `next` (which, again, if not for `SQLFreeStmt`, would be valid).